src/main/java/com/minecolonies/coremod/colony/Colony.java

```diff
@@ -441,46 +509,190 @@ public int getID()
+                    break;
+                case "diamonds" : this.triggerAchievement(ModAchievements.achievementMine100Diamonds);
+                    break;
+                case "huts"     : this.triggerAchievement(ModAchievements.achievementBuild100Huts);
+                    break;
+                case "fish"     : this.triggerAchievement(ModAchievements.achievementCatch100Fish);
+                    break;
+                case "wheat"    : this.triggerAchievement(ModAchievements.achievementHarvest100Wheat);
+                    break;
+                case "potatoes" : this.triggerAchievement(ModAchievements.achievementHarvest100Potatoes);
+                    break;
+                case "carrots"  : this.triggerAchievement(ModAchievements.achievementHarvest100Carrots);
+                    break;
+                case "saplings" : this.triggerAchievement(ModAchievements.achievementPlant100Saplings);
+                    break;
+                case "trees"    : this.triggerAchievement(ModAchievements.achievementFell100Trees);
+                    break;
+            }
         }
-        if (mobKills >= NUM_MOBS_ACHIEVEMENT_FOURTH)
+        if (statisticAmount >= NUM_ACHIEVEMENT_FOURTH)
         {
-            this.triggerAchievement(ModAchievements.achievementKill500Mobs);
+            switch (statistic)
+            {
+                case "mobs"     : this.triggerAchievement(ModAchievements.achievementKill500Mobs);
+                    break;
+                case "ores"     : this.triggerAchievement(ModAchievements.achievementMine500Ores);
+                    break;
+                case "diamonds" : this.triggerAchievement(ModAchievements.achievementMine500Diamonds);
+                    break;
+                case "huts"     : this.triggerAchievement(ModAchievements.achievementBuild500Huts);
+                    break;
+                case "fish"     : this.triggerAchievement(ModAchievements.achievementCatch500Fish);
+                    break;
+                case "wheat"    : this.triggerAchievement(ModAchievements.achievementHarvest500Wheat);
+                    break;
+                case "potatoes" : this.triggerAchievement(ModAchievements.achievementHarvest500Potatoes);
+                    break;
+                case "carrots"  : this.triggerAchievement(ModAchievements.achievementHarvest500Carrots);
+                    break;
+                case "saplings" : this.triggerAchievement(ModAchievements.achievementPlant500Saplings);
+                    break;
+                case "trees"    : this.triggerAchievement(ModAchievements.achievementFell500Trees);
+                    break;
+            }
         }
-        if (mobKills >= NUM_MOBS_ACHIEVEMENT_FIFTH)
+        if (statisticAmount >= NUM_ACHIEVEMENT_FIFTH)
         {
-            this.triggerAchievement(ModAchievements.achievementKill1000Mobs);
+            switch (statistic)
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Add a default case to this switch. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3ASwitchLastCaseIsDefaultCheck)